### PR TITLE
Fix/inherit config bugs

### PR DESCRIPTION
**Pull request by XIEZHENGYAO**
### Issues associated with this PR

https://github.com/mosn/mosn/issues/1816

### Solutions
Remove code: 
``` golang
c = configmanager.Load(configmanager.GetConfigPath())
```

### CODE REVIEW

File: pkg/mosn/mosn.go [hunk 1]
```diff
@@ -87,8 +87,6 @@ func (m *Mosn) upgradeCheck() {
 		log.StartLogger.Infof("[mosn] [NewMosn] active reconfiguring")
 		// set Mosn Active_Reconfiguring
 		store.SetMosnState(store.Active_Reconfiguring)
-		// parse MOSNConfig again
-		c = configmanager.Load(configmanager.GetConfigPath())
```
Comment (nejisama) [inline pkg/mosn/mosn.go:91]: keep this，use new c to m.Config

Comment (XIEZHENGYAO) [reply pkg/mosn/mosn.go]: Done, plz review again.

Comment (taoyuanyuan) [reply pkg/mosn/mosn.go]: We should reset `m.config` directly whether or not `InheritOldMosnconfig` is true.

Comment (XIEZHENGYAO) [reply pkg/mosn/mosn.go]: Done.

**Review by taoyuanyuan — APPROVED** [review]
LGTM

**Review by nejisama — APPROVED** [review]

**Pull request merged**